Replicator/CookieStore.cc

```diff
@@ -36,8 +36,54 @@ using namespace fleece;
 
 namespace litecore { namespace repl {
 
+    static void cap_to_max(struct tm* inputTime) {
+        Warn("Received a struct tm that overflows 32-bit time_t!  Capping the value...");
+        inputTime->tm_year = 138;
+        inputTime->tm_mon = 0;
+        inputTime->tm_mday = 19;
+        inputTime->tm_yday = 19;
+        inputTime->tm_wday = 2;
+        inputTime->tm_hour = 3;
+        inputTime->tm_min = 14;
+        inputTime->tm_sec = 7;
+        inputTime->tm_isdst = 0;
+    }
+
+    static void adjust_for_overflow(struct tm* inputTime)
+    {
+        if(sizeof(time_t) > 4 || inputTime->tm_year < 138) {
+            return;
+        }
+
+        if(inputTime->tm_mon > 0) {
```

Comment:
Please combine all five of these tests into one with `||`. Then since there's only one call to `cap_to_max`, you can move that into the body of the `if`.
(Sorry to be picky, but this code can be so much smaller without losing clarity.)

Reply:
I wanted to do that but this is not just one condition.  For example, I only care if the seconds are greater than 7 if the minutes are <= 14.  Otherwise any time with seconds greater then 7 will get capped.  Am I having a brain fart?

Reply:
This just made me realize that what I wrote doesn't actually achieve what I want...hold on that